src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -193,7 +195,16 @@ public static string GenerateNameFromType(this SemanticModel semanticModel, ITyp
 
             // Otherwise assume no pluralization, e.g. using 'immutableArray', 'list', etc. instead of their
             // plural forms
-            return type.CreateParameterName(capitalize);
+            if (type.IsSpecialType() ||
+                type.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T ||
+                type.TypeKind == TypeKind.Pointer)
```

Comment:
curious why pointer?

Reply:
The same reason as `Nullabel<T>`. There were some test failing fot `int?` - suggested `nullable` and `int*` - suggested `pointer` while it shouldn't suggest anything for this cases